docs/xks/developer-guide/scheduling.md

```diff
@@ -0,0 +1,138 @@
+          - key: label-key
+            operator: In
+            values:
+            - label-value
+```
+
+This example only allow pods to be scheduled on nodes with a key `kubernetes.io/name` with value `ABC` or `XYZ` Among the nodes matching this criteria, nodes with key `label-key` and value `label-value` will be preferred.
+
+The `weight` field ir ranged 1-100 and for each node matching all scheduling requirements, the kube-scheduler computes a score, as mentioned earlier. Then adds this number to that sum to calculate the best matching node.
+
+### podAffinity and podAntiaffinity
+
+podAffinity and podAntiAffinity lets you constrain which nodes pod are eligible to be scheduled on based of label of the pods running on the node rather then labels on the node.
+
+podAnitAffinity
+
+```yaml
+spec:
+  affinity:
+    podAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
+      - labelSelector:
+          matchExpressions:
+          - key: label1
+            operator: In
+            values:
+            - label-value
+        topologyKey: topology.kubernetes.io/zone
+    podAntiAffinity:
+      preferredDuringSchedulingIgnoredDuringExecution:
+      - weight: 100
+        podAffinityTerm:
+          labelSelector:
+            matchExpressions:
+            - key: label2
+              operator: In
+              values:
+              - label-value-anti
+          topologyKey: topology.kubernetes.io/zone
+```
+
+This example show an example where we use both affinity rules.
+
+Affinity rule: the pod can be only scheduled onto a node if that node is in the same zone as at least one already-running pod that has a label with key `label1` and value `label-value`.
+
+antiAffinity rule: the pod should not be scheduled onto a node if that node is in the same zone as a pod with label having key `label2` and value `label-value-anti`
+
+```yaml
+  affinity:
+    podAntiAffinity:
+      preferredDuringSchedulingIgnoredDuringExecution:
+        - podAffinityTerm:
+            labelSelector:
+              matchExpressions:
+                - key: prometheus
+                  operator: In
+                  values:
+                    - xks
+            topologyKey: kubernetes.io/hostname
+          weight: 100
+        - podAffinityTerm:
+            labelSelector:
+              matchExpressions:
+                - key: prometheus
+                  operator: In
+                  values:
+                    - xks
+            topologyKey: topology.kubernetes.io/zone
+          weight: 100
+```
+
+This is an example configuration of podAntiAffinity for Prometheus. Spreading the pod deployment based on `topology.kubernetes.io/zone` and `topology.kubernetes.io/hostname` to only allow 1 pod on each node and to mitigate downtime in case an entire zone goes down, i.e if a pod runs in zone A with key `prometheus` and value `xks` dont schedule in zone A, choose zone B or C. Note that these settings are "preferred" and not required.

```

Comment:
```suggestion
This is an example configuration of podAntiAffinity for Prometheus. Spreading the pod deployment based on `topology.kubernetes.io/zone` and `topology.kubernetes.io/hostname` to only allow 1 pod on each node and to mitigate downtime in case an entire zone goes down, e.g: if a pod runs in zone A with key `prometheus` and value `xks` do not schedule it in zone A - choose zone B or C. Note that these settings are "preferred" and not required.
```